File: year 2 - AP Comp Sci/labs/arrayfunhouse1/ArrayFunHouse.java
public class ArrayFunHouse
{
	//instance variables and constructors could be used, but are not really needed

	//getSum() will return the sum of the numbers from start to stop, not including stop
	//Use a for loop to traverse the array
	public static int getSum(int[] numArray, int start, int stop)
	{
		int result = 0;
		for(int i = start; i < stop; i++)
		{
			result += numArray[i];
		}

		return result;
	}

	//getCount() will return number of times val is present
	public static int getCount(int[] numArray, int val)
	{
		int occurences = 0;
		for(int i = 0; i < numArray.length; i++)
		{
			if(numArray[i] == val) {
				occurences++;
			}
		}
		return occurences;
	}

	//removes all instances of val?
	public static int[] removeVal(int[] numArray, int val)
	{

		int occurences = getCount(numArray, val);
		int[] result = new int[numArray.length-occurences];
		int resultPlace = 0;

		for(int i = 0; i < numArray.length; i++)
		{
			if(numArray[i] != val) {
				result[resultPlace] = numArray[i];
				resultPlace++;
			}
		}

		return result;
	}
}

File: year 2 - AP Comp Sci/labs/arrayfunhouse1/ArrayFunHouse.rb
class ArrayFunHouse
    def self.get_sum(array, start, stop)
        sum = 0.0
        if array.length > 0 then
            for i in start..stop
                sum += array[i]
            end
        end
        return sum
    end
    def self.get_count(array, val)
        count = 0
        if array.length > 0
            array.each do |i|
                if i == val
                    count += 1
                end
            end
        end
        return count
    end
    def self.remove_val(array,val)
        arr = array.clone
        arr.delete_at(arr.find_index(val))
        return arr
    end
end

if __FILE__ == $0
    one = [4,10,0,1,7,6,5,3,2,9]
    puts "#{one}"
    puts "sum of spots 3-6 = #{ArrayFunHouse.get_sum(one,3,6)}"
    puts "sum of spots 2-9 = #{ArrayFunHouse.get_sum(one,2,9)}"
    puts "# of 4s = #{ArrayFunHouse.get_count(one,4)}"
    puts "# of 7s = #{ArrayFunHouse.get_count(one,7)}"
    puts "before removing all 7s #{one}"
    one = ArrayFunHouse.remove_val(one,7)
    puts "after removing all 7s #{one}"
    puts "# of 7s = #{ArrayFunHouse.get_count(one,7)}"
    puts "********************************************"

    two = [4,2,3,4,6,7,8,9,0,10,0,1,7,6,5,3,2,9,9,8,7]
    puts "#{two}"
    puts "sum of spots 20-21 = #{ArrayFunHouse.get_sum(two,20,21)}"
    puts "sum of spots 8-11 = #{ArrayFunHouse.get_sum(two,8,11)}"
    puts "# of 2s = #{ArrayFunHouse.get_count(two,2)}"
    puts "# of 9s = #{ArrayFunHouse.get_count(two,9)}"
    puts "before removing all 9s #{two}"
    two = ArrayFunHouse.remove_val(two,9)
    puts "after removing all 9s #{two}"
    puts "# of 7s = #{ArrayFunHouse.get_count(two,9)}"
end
